src/Control/Middleware/CanonicalURLMiddleware.php

```diff
@@ -17,10 +18,31 @@
  *  - redirect basic auth requests to HTTPS
  *  - force WWW, redirect to the subdomain "www."
  *  - force SSL, redirect to https
+ *  - force the correct path (with vs without trailing slash)
  */
 class CanonicalURLMiddleware implements HTTPMiddleware
 {
     use Injectable;
+    use Configurable;
+
+    /**
+     * If set, the trailing slash configuration set in {@link Controller::add_trailing_slash} is enforced
+     * with a redirect.
+     */
+    protected bool $forceTrailingSlash = true;
```

Comment:
This is a badly named property (along with all the other "forceTrailingSlash*" properties.)  It does not "force tailing slash" - instead it is basically "do not ignore the trailing slash config on Controller".  The naming here makes this very confusing

We'll need to rename this, perhaps simply to "respectTrailingSlashConfig*"? Though there's probably something more better and more concise that would be used.  Another option would be to rename this to "ignoreTrailingSlashConfig*" and then reverse the logic

Reply:
How about `redirectWithTrailingSlashConfig`? A bit verbose but it makes it clear, I think?

Reply:
hm. You're right it's not forcing the trailing slash, but rather its config value. The correct name would then be "forceTrailingSlashConfig"?

Reply:
I think 'force' is probably the wrong word here, though I think just 'enforce' works fine

So this would become `enforceTrailingSlashConfig`

Does that sound alright?